### ISSUE TITLE
Return no collision if security_margin is -infinity

### ISSUE THREAD

**Issue by florent-lamiraux** [humanoid-path-planner/hpp-fcl#269]
https://github.com/humanoid-path-planner/hpp-fcl/blob/47941ac4cdb8c72c2640babd73f39812946ae302/src/collision.cpp#L166
I would like to add the following test before the above line
```
if (request.security_margin == -std::numeric_limits<FCL_REAL>::infinity()){
  result.clear();
  return false;
}  
```
so that the pair is considered not colliding without any further test.
Do you think
  1. that it is a bad idea,
  2. that testing against `-std::numeric_limits<FCL_REAL>::infinity()` is not robust ?

Any feedback is welcome.

**Comment by florent-lamiraux** [comment]
I realize though that passing `std::numeric_limits<FCL_REAL>::infinity()` through python bindings may  be risky.
Perhaps testing against `-1e20` is enough.

**Comment by jmirabel** [comment]
Is a negative security margin supported at the moment ? I don't think it works for all the pairs (especially when it involves BVH).

My mind is not set on this. I think there is a cleaner way of doing this. The bound can be computed using the two bounding box (eventually converted to sphere to simplify computations) and there relative pose.

**Comment by florent-lamiraux** [comment]
Negative security margins are not supported for the moment. I just wish to disable a test by setting the security margin to -infinity.
I do not understand your comment.

**Comment by jmirabel** [comment]
> I do not understand your comment.

I meant:

The collision check consist in checking `signed_distance > security_margin`. For two spheres of diameter 1, necessarily, `signed_distance > -2` so if `security_margin < -2` you can say there is no collision without further tests. Considering a bounding sphere of each object, you can find an upper bound for `security_margin`. This tends to give negative security margins a meaning.

> I just wish to disable a test by setting the security margin to -infinity.

This can be convenient (for testing only, I don't advise it for production) and is consistent theoretically. Using `numpy.inf` or `float('inf')` allows to use infinity from Python so the equality shouldn't be a problem. Alternatively, you can use `security_margin <= std::numeric_limits<FCL_REAL>::max()`, or just `security_margin < 0`.

**Comment by jcarpent** [comment]
Solved by #271
